api/server/server.go

```diff
@@ -99,12 +100,8 @@ func (s *Server) ServeApi(protoAddrs []string) error {
 				chErrors <- err
 				return
 			}
-			s.eng.OnShutdown(func() {
-				if err := srv.Close(); err != nil {
-					logrus.Error(err)
-				}
-			})
-			if err = srv.Serve(); err != nil && strings.Contains(err.Error(), "use of closed network connection") {
+			s.servers = append(s.servers, srv)
+			if err := srv.Serve(); err != nil && strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
You should wait for some time until ALL servers will try to close. This will return on first error IIRC.